### ISSUE TITLE
[HOLD for payment 2023-04-05] [HOLD][$2000] https://staging.new.expensify.com/details crashed app

### ISSUE THREAD

**Issue by mallenexpensify** [Expensify/App#15486]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
Was testing something and clicked on https://staging.new.expensify.com/details  

## Expected Result:
Our standard error page would show
https://www.screencast.com/t/cjYv4o0xprO

## Actual Result:
The site/app crashed
https://www.screencast.com/t/CZWzKrlv7

## Workaround:
N/A

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [x] iOS / Firefox (and likely more)
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** v1.2.75-0
**Reproducible in staging?:** Staging
**Reproducible in production?:** yes
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):** manager.allen@expensicorp.com 
**Logs:**   https://www.screencast.com/t/tN52lVXhLFS3
 
[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019da6596d73718432</li>
        <li>Upwork Job ID: 1630735661398630400</li>
        <li>Last Price Increase: 2023-03-08</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Current assignee @mallenexpensify is eligible for the Bug assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
@marcochavezf, @mallenexpensify Uh oh! This issue is overdue by 2 days. Don't forget to update your issues!

**Comment by marcochavezf** [comment]
No update yet, this is my priority for tomorrow

**Comment by marcochavezf** [comment]
This issue was [fixed](https://github.com/Expensify/App/pull/10820) before.

**Comment by marcochavezf** [comment]
Focused on the LeaveRoom command, making it External meanwhile in case someone wants to propose a solution.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~019da6596d73718432

**Comment by MelvinBot** [comment]
Current assignee @mallenexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @mananjadhav (`External`)

**Comment by MelvinBot** [comment]
Current assignee @marcochavezf is eligible for the External assigner, not assigning anyone new.

**Comment by situchan** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App crashes when navigate to /details page directly from browser address bar

### What is the root cause of that problem?
`DetailsPage` is expected to have query params like login or reportID
i.e. https://staging.new.expensify.com/details?login=situchan%40gmail.com
But if we just enter `/details` without any param, `this.props.route.params` returns `undefined`
So it crashes on 2 parts:
1.
https://github.com/Expensify/App/blob/44c19cbd1891a255394f984bab4b815c580c9e13/src/pages/DetailsPage.js#L96
this calls
```
    removeSMSDomain(text) {
        return text.replace(REMOVE_SMS_DOMAIN_PATTERN, '');
    },
```
in expensify-common repo.
Here, text is `undefined` which makes crash

2.
https://github.com/Expensify/App/blob/44c19cbd1891a255394f984bab4b815c580c9e13/src/pages/DetailsPage.js#L100
I think this crash was introduced in #12364 which shows placeholder page instead of full screen loading.

### What changes do you think we should make in order to solve the problem?
1. We should fallback to `''` in lodashGet 3rd param
https://github.com/Expensify/App/blob/44c19cbd1891a255394f984bab4b815c580c9e13/src/pages/DetailsPage.js#L88
2. We should use `lodashGet` here
https://github.com/Expensify/App/blob/44c19cbd1891a255394f984bab4b815c580c9e13/src/pages/DetailsPage.js#L100

**Comment by priyeshshah11** [comment]
## Proposal
### Please re-state the problem that we are trying to solve in this issue.
App crashes when navigate to /details page directly from browser address bar

### What is the root cause of that problem?
It crashes in the `removeSMSDomain` function where `text` is passed as undefined & then we try to do a replace on that. This crash sometimes happens in other parts of the app too, not just details page.

### What changes do you think we should make in order to solve the problem? 
In `removeSMSDomain` check for a non-null/undefined value before calling the replace function on it.

**Comment by situchan** [comment]
> This crash sometimes happens in other parts of the app too, not just details page.

I think it's better to leave this for tracking purpose than hiding bug. If we add safety check in `removeSMSDomain`, it will be difficult to find the root cause of unexpected behavior in the future since crash won't happen.

**Comment by tienifr** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App crashes when navigating to `/details`

### What is the root cause of that problem?
We're not setting the default value for the `login` params. The `login` params is required for operations within the `DetailsPage` like `Str.isSMSLogin`. And if `login` is undefined, it will crash the app.

Also we're expecting the `params` to be defined as in `this.props.route.params.reportID`, it will also cause the crash here.

### What changes do you think we should make in order to solve the problem?

When we're using required string props inside the page, we usually provide a default empty string value for it to make sure it doesn't crash the app. For example in here https://github.com/Expensify/App/blob/0e287b25e3453b5ef0cc1f197e7a38919c0f667c/src/components/ConfirmView.js#L47.

We need to do the same thing for the navigation params passed in the page as well. We can do this by adding 
```
initialParams: {
        login: '',
        reportID: '',
    }
```
in https://github.com/Expensify/App/blob/6dc3e818e021d69b9d83bba2f59363552c1b8455/src/libs/Navigation/AppNavigator/ModalStackNavigators.js#L132
where we initialize the `DetailsPage` screen in the navigator.

### What alternative solutions did you explore? (Optional)
We can add `''` as the fallback value for login inside the `render` method in `DetailsPage`, but I don't prefer it because it's more of a case-by-case fix (not general) and is not consistent with how we're handling default value for string props in the page.

Imagine if this happens in another page, we can also just add the `initialParams` in the navigator rather than adding the fallback `''` when we use the param on a case-by-case basis inside the page.

**Comment by MelvinBot** [comment]
Looks like something related to `react-navigation` may have been mentioned in this issue discussion.

As a reminder, please make sure that all proposals are not workarounds and that any and all attempt to fix the issue holistically have been made before proceeding with a solution. Proposals to change our `DeprecatedCustomActions.js` files should not be accepted.

Feel free to drop a note in [#expensify-open-source](https://expensify.slack.com/archives/C01GTK53T8Q) with any questions.

**Comment by nishchay7pixels** [comment]
# Proposal

### Please re-state the problem that we are trying to solve in this issue.

App crashes when someone will navigate to /details page without actually passing any parameter.

### What is the root cause of that problem?

When redirecting to /details applicationexpect you to also pass query params in the url which are later accessed in DetailsPage.js

```
render(){
let details = lodashGet(this.props.personalDetails, lodashGet(this.props.route.params, 'login'));

        if (!details) {
            const login = lodashGet(this.props.route.params, 'login');
            details = {
                login,
                displayName: ReportUtils.getDisplayNameForParticipant(login),
                avatar: ReportUtils.getAvatar(lodashGet(details, 'avatar', ''), login),
            };
        }
..
```

### What changes do you think we should make in order to solve the problem?

This can be solved by adding a conditional check inside the DetailsPage screen to ensure that the required params are defined before using them. 
For eg,
```
render() {
        if (!this.props.route.params || !this.props.route.params.login || !this.props.route.params.reportID) {
            return null; // or show an error message
        }
        let details = lodashGet(this.props.personalDetails, lodashGet(this.props.route.params, 'login'));

        if (!details) {
            const login = lodashGet(this.props.route.params, 'login');
            details = {
                login,
                displayName: ReportUtils.getDisplayNameForParticipant(login),
                avatar: ReportUtils.getAvatar(lodashGet(details, 'avatar', ''), login),
            };
        }
...
```

**Comment by bernhardoj** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The app is crashed when we access details page with no login params. We want to show not found page when that happen.

### What is the root cause of that problem?
When we open the page without login params, we dismiss it inside `componentDidMount`. However, this does not prevent the component to be rendered. Inside the render, we access the login params and pass it to `Str.isSMSLogin` which will call `replace` function on the passed variable (which is the undefined login).

https://github.com/Expensify/App/blob/edb7bc9ebaaeabe70f599f152a319e64ab7a5327/src/pages/DetailsPage.js#L96

### What changes do you think we should make in order to solve the problem?

1. Remove `componentDidMount`
2. Inside `render` get the login params and set an empty string as the default value and store it into a new variable called `login`. This new variable will be used to every place that we currently use.
3. 
https://github.com/Expensify/App/blob/edb7bc9ebaaeabe70f599f152a319e64ab7a5327/src/pages/DetailsPage.js#L85-L94

4. Inside `render` get the reportID params and set an empty string as the default value and store it into a new variable called `reportID` and pass it to `shouldShowBackButton = Boolean(reportID)`. 
5. Wrap the component (as `ScreenWrapper` children) with `FullPageNotFoundView` and set the `shouldShow` with `Boolean(login)`

**Comment by tienifr** [comment]
> I think we should show the error page as that's what's shown when you go to a random URL like

@mallenexpensify can you confirm if this should be a full-screen error page or it just shows in the right side modal like this?
<img width="965" alt="Screenshot 2023-03-06 at 22 32 01" src="https://user-images.githubusercontent.com/113963320/223155903-f6e46208-60f5-4dcc-bd47-fb8ea8920c7f.png">

**Comment by MelvinBot** [comment]
@mananjadhav, @marcochavezf, @mallenexpensify Whoops! This issue is 2 days overdue. Let's get this updated quick!

**Comment by mallenexpensify** [comment]
Checking on the expected behavior [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1678145945634149) 
with the goal to update the [testing guidelines ](https://docs.google.com/document/d/1UaWlcieiwLnsKfeKw7iwLF7vgYRmdw_-K8OcY5aPBRU/edit#) once I find out.

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
App crashes on navigating the `/details` page without query param

### What is the root cause of that problem?
As we are expecting the query param `login` for the details page and on missing of the same it will try to operate string function(replace) on `undefined` which is causing the issue.

### What changes do you think we should make in order to solve the problem?
Check the `login` query param first in the render method and in case of unavailable make the `return` statement which consists of the `ScreenWrapper` wrapping the `FullPageNotFoundView` component which will help to prevent unwanted further execution.
https://github.com/Expensify/App/blob/6b2a7690ee170994f1e533473b88fb4802957b54/src/pages/DetailsPage.js#L85

With that, `componentDidMount` won't be required as it is dismissing the modal which is not the requirement now.
We can take the below line to the first place to decide whether to show back button or not for 'not found page'.
https://github.com/Expensify/App/blob/6b2a7690ee170994f1e533473b88fb4802957b54/src/pages/DetailsPage.js#L100

Above changes will show the `Not found page` on the right pane.

**Comment by mallenexpensify** [comment]
> @mallenexpensify can you confirm if this should be a full-screen error page or it just shows in the right side modal like this?

@tienifr the error page should cover the main chat window on Web/Desktop (assuming it's not minimized) and the full page on mobile. 
<img width="1385" alt="image" src="https://user-images.githubusercontent.com/22508304/223895410-09943879-f6b7-493d-8621-db85ab132870.png">

**Comment by bernhardoj** [comment]
@mallenexpensify I'm afraid it's not possible to show it like that. The SS you provided is a not found chat and we need to be on the chat/report screen to show that, but if we open /details, we will be at the details screen.

Report screen -> Details screen (we are here).

We can either show a full page not found or just at the RHN. Looking at the other RHN screen, they all show not found page at the RHN (example https://staging.new.expensify.com/r/1/details)

**Comment by situchan** [comment]
https://staging.new.expensify.com/details/random shows not found page on full screen
![image](https://user-images.githubusercontent.com/108292595/223905813-cc38f50e-81fa-415e-8cb1-59c17ddeb8fb.png)


https://staging.new.expensify.com/details?login=test shows not found page on modal (Detail page)
![image](https://user-images.githubusercontent.com/108292595/223905723-f247b326-79cd-477c-8b5e-f41290a08d61.png)

2nd one is quick and easy to implement

**Comment by Pujan92** [comment]
@mallenexpensify  for the `/details` route we already reached the right pane and there only we are validating data to decide whether to show `not found page` or not which ultimately shows that page in the right pane.

**Comment by mallenexpensify** [comment]
Thanks @bernhardoj and @Pujan92 .   If we're unable to show the error message in the main chat window, let's do RHP/RHN

**Comment by mallenexpensify** [comment]
@mananjadhav or @marcochavezf , if you have time can you review the above proposals?  You both might be traveling the next few days though so, apologies for any delays for the reviews

**Comment by MelvinBot** [comment]
@mananjadhav @marcochavezf @mallenexpensify this issue was created 2 weeks ago. Are we close to approving a proposal? If not, what's blocking us from getting this issue assigned? Don't hesitate to create a thread in #expensify-open-source to align faster in real time. Thanks!

**Comment by mallenexpensify** [comment]
Put on hold as ExpensiConX is this week and it might take a bit to review the above

**Comment by mananjadhav** [comment]
@mallenexpensify A bit confused here with all the comments, is the GH body the right expected result? Is this what should I assume the [expected result to be](https://github.com/Expensify/App/issues/15486#issuecomment-1462596953)?

**Comment by situchan** [comment]
> With @bernhardoj mentioning `FullPageNotFoundView` first in the proposal I am going ahead with their proposals.

I first proposed this on Mar 3 10:51 CST
@bernhardoj proposed this on Mar 3 10:57 CST

**Comment by mananjadhav** [comment]
I can't make out from the edits which one was first.  The first version if your proposal doesn't mention it, where as @bernhardoj's mentions it.

> A quick note to the contributors, if you're adding any revisions of the proposals that update the logic of the proposed solution, I would recommend sending it a new comment

and hence I explicitly highlighted this.

**Comment by situchan** [comment]
Oh, there's no guideline about that. I noted it now and whenever I find similar case where edited proposal (without new comment with "Proposal updated") was accepted, I will report that.

**Comment by marcochavezf** [comment]
> Oh, there's no guideline about that. I noted it now and whenever I find similar case where edited proposal (without new comment with "Proposal updated") was accepted, I will report that.

Yeah, I think we'd need to update the guidelines about the edited proposals. Thanks for noticing that.

**Comment by marcochavezf** [comment]
> With @bernhardoj mentioning FullPageNotFoundView first in the proposal I am going ahead with their proposals.

Thanks @mananjadhav also the proposal LGTM. Assigning @bernhardoj 🚀

**Comment by bernhardoj** [comment]
PR is ready for review.

cc: @mananjadhav

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@mananjadhav / @marcochavezf] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@mananjadhav / @marcochavezf] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@mananjadhav / @marcochavezf] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@mallenexpensify] Determine if we should create a regression test for this bug.
- [ ] [@mananjadhav] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@mallenexpensify] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by marcochavezf** [comment]
Seems this was an unhandled edge case, and I don't think was introduced by another PR. Also, the regression tests will be enough to prevent this bug in the future, no need to update the checklist here.

**Comment by mananjadhav** [comment]
Agreed with your assessment @marcochavezf.

@mallenexpensify This is due for payout. Can you please help?

**Comment by mallenexpensify** [comment]
Thanks @mananjadhav.  Can you and @bernhardoj can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~01512148a62908748b

It doesn't appear a bonus is due as there were 6 days between hire and merge.

**Comment by bernhardoj** [comment]
Accepted.

Can we take @mananjadhav approval as the merge date, because the next few days is just some discussion that requires no changes to the PR?

**Comment by mananjadhav** [comment]
Yeah I think we finished the checklist within a day, and the discussion that happened didn't have any rework/changes. I think the bonus should apply.

@marcochavezf what do you think ?

**Comment by mallenexpensify** [comment]
@bernhardoj and @mananjadhav paid $2000 each. 
I've added this issue to a new `Edge Cases` section of the Testing Guidelines doc - [link](https://docs.google.com/document/d/1UaWlcieiwLnsKfeKw7iwLF7vgYRmdw_-K8OcY5aPBRU/edit#heading=h.aurxvmlpvqou)

**Comment by mananjadhav** [comment]
@mallenexpensify did you get a chance to look this [comment](https://github.com/Expensify/App/issues/15486#issuecomment-1500784281) ?

**Comment by mallenexpensify** [comment]
Thanks @mananjadhav , I missed that, sorry. 
Usually it'd be the engineer merging the PR that is the trigger.  So, you're saying that the PR was _cool_ but you chatted about for a while to make sure it was good, then Marco 👍 it?  

My main concern here is we don't want to set precedence where others reference this issue/PR and say... "it happened here, therefore I should get the bonus to".  

Let's let @marcochavezf chime in

**Comment by mananjadhav** [comment]
Thanks for o

**Comment by marcochavezf** [comment]
Yeah, we're ok with the bonus here. The delay here was caused by me, and @mananjadhav is correct about a post-implementation discussion that didn't happen and also delayed the merge (but the PR was already reviewed and tested within a day).
